figures/create-endpoint-activity.mmd

```diff
@@ -0,0 +1,21 @@
+stateDiagram-v2
+
+    state "Client hits /create endpoint" as conn 
+    state "Client tries to edit room parameters" as param
+    state fork_state <<fork>>
+    state "Title greater than 255 char" as title
+    state "Name is less than 3 char or greater than 32 char" as name
+    state "Visibility is not an allowed value" as visibility
+    state join_state <<fork>>
+    [*] --> conn: Client connects.
+    conn --> param: 
+    param --> fork_state
+    fork_state --> title: Title is too big.
+    fork_state --> name: Name too small or big.
+    fork_state --> visibility: Visibility is not public, private, or unlisted.
+    title --> join_state
```

Comment:
This is fine I guess, but I think it's too specific. If the validation rules change, then this diagram becomes out of date.